Core/Core/Assignments/AssignmentList/ViewModel/AssignmentListViewModel.swift

```diff
@@ -101,14 +107,21 @@ public class AssignmentListViewModel: ObservableObject {
     ) {
         self.userDefaults = userDefaults
         self.courseID = context.id
-        self.defaultGradingPeriod = defaultGradingPeriod
-        self.selectedGradingPeriod = self.defaultGradingPeriod
 
         featureFlags.refresh()
     }
 
     // MARK: - Functions
 
+    public func viewDidAppear() {
+        loadAssignmentListPreferences()
+        course.refresh(force: true)
+        gradingPeriods.refresh(force: true)
+        filterOptionsDidUpdate(filterOptions: selectedFilterOptions, gradingPeriod: defaultGradingPeriod)
+
+        isFilterIconSolid = ![0, AssignmentFilterOption.allCases.count].contains(selectedFilterOptions.count)
```

Comment:
**Optional.**
Another (more affirmative) way of doing this check:
(1 ..< AssignmentFilterOption.allCases.count).contains(selectedFilterOptions.count)

As, it is quite less readable to use `not` operation to express exclusivity logic.